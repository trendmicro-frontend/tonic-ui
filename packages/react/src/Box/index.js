import styled from '@emotion/styled';
import isPropValid from '@emotion/is-prop-valid';
import { ensureArray } from 'ensure-type';
import { cx, system } from '../shared/styled-system';

const shouldForwardProp = (() => {
  const stylePropMap = ensureArray(system.propNames)
    .reduce((acc, val) => {
      acc[val] = true;
      return acc;
    }, {});
  const omittedStylePropMap = {
    ...stylePropMap,

    // The `as` prop is supported by Emotion
    'as': true,
  };

  return prop => isPropValid(prop) && !omittedStylePropMap[prop];
})();

const createPseudoClassTransformFunction = name => prop => {
  const result = {};
  if (Array.isArray(prop)) {
    result[`${name}(${prop[0]})`] = prop[1];
  } else if (typeof prop === 'object') {
    for (const key in prop) {
      if (Object.prototype.hasOwnProperty.call(prop, key)) {
        result[`${name}(${key})`] = prop[key];
      }
    }
  }
  return result;
};

/**
 * Pseudo-classes
 * https://developer.mozilla.org/en-US/docs/Web/CSS/
 */

const active = [
  '&:active',
  '&[data-active=true]'
].join(',');
const checked = [
  '&[aria-checked=true]',
  '&:checked',
].join(',');
const selected = [
  '&[aria-selected=true]',
  '&[data-selected=true]',
  '&:active[data-selected=true]',
  '&:hover[data-selected=true]',
].join(',');
const disabled = [
  '&[aria-disabled=true]',
  '&:disabled',
  '&:disabled:hover',
  '&:disabled:focus',
  '&:hover[aria-disabled=true]',
  '&:focus[aria-disabled=true]',
].join(',');
const empty = '&:empty';
const enabled = [
  '&:enabled',
  '&:enabled:focus',
  '&:enabled:hover',
].join(',');
const firstChild = '&:first-child';
const firstOfType = '&:first-of-type';
const fullscreen = '&:fullscreen';
const focus = '&:focus';
const focusActive = '&:focus:active';
const focusHover = '&:focus:hover';
const focusWithin = '&:focus-within';
const focusSelected = ['&[aria-selected=true]:focus', '&[data-selected=true]:focus'].join(',');
const hover = '&:hover';
const indeterminate = '&:indeterminate';
const valid = [
  '&[aria-invalid=false]',
  '&:valid',
].join(',');
const invalid = [
  '&[aria-invalid=true]',
  '&:invalid',
].join(',');
const lastChild = '&:last-child';
const lastOfType = '&:last-of-type';
const nthOfTypeFn = createPseudoClassTransformFunction('&:nth-of-type');
const readOnly = [
  '&[aria-readonly=true]',
  '&:read-only',
].join(',');
const visited = '&:visited';

/**
 * Pseudo-elements
 * https://developer.mozilla.org/en-US/docs/Web/CSS/
 */

const after = '&::after';
const backdrop = '&::backdrop';
const before = '&::before';
const cue = '&::cue';
const firstLetter = '&::first-letter';
const firstLine = '&::first-line';
const placeholder = '&::placeholder';
const selection = '&::selection';

const _Box = styled('div', {
  shouldForwardProp,
})(system);

const Box = styled(_Box)(
  ({
    // pseudo-classes
    _active,
    _checked,
    _selected,
    _disabled,
    _empty,
    _enabled,
    _firstChild,
    _firstOfType,
    _fullscreen,
    _focus,
    _focusActive,
    _focusHover,
    _focusWithin,
    _focusSelected,
    _hover,
    _indeterminate,
    _valid,
    _invalid,
    _lastChild,
    _lastOfType,
    _nthOfType,
    _readOnly,
    _visited,

    // pseudo-elements
    __after,
    __backdrop,
    __before,
    __cue,
    __firstLetter,
    __firstLine,
    __placeholder,
    __selection,
  }) => {
    let rest = null;

    if (_nthOfType) {
      rest = { ...rest, ...nthOfTypeFn(_nthOfType) };
    }

    return cx({
      /**
       * Pseudo-classes must be declared in a specific order, as shown below:
       *
       * ```
       * :link
       * :visited
       * :hover
       * :active
       * ```
       *
       * Each pseudo-class corresponds to an event which can only happen later in the timeline than the one before.
       *
       * That is to say:
       *
       * 1. A link is unvisited before it is visited.
       * 2. A link is visited before it is hovered over.
       * 3. A link is hovered over before it is in active use.
       */
      [focus]: _focus,
      [visited]: _visited,
      [hover]: _hover,
      [focusHover]: _focusHover,
      [active]: _active,
      [focusActive]: _focusActive,
      [focusSelected]: _focusSelected,
      [focusWithin]: _focusWithin,
      [checked]: _checked,
      [selected]: _selected,
      [disabled]: _disabled,
      [empty]: _empty,
      [enabled]: _enabled,
      [firstChild]: _firstChild,
      [firstOfType]: _firstOfType,
      [fullscreen]: _fullscreen,
      [indeterminate]: _indeterminate,
      [valid]: _valid,
      [invalid]: _invalid,
      [lastChild]: _lastChild,
      [lastOfType]: _lastOfType,
      [readOnly]: _readOnly,

      // pseudo-elements
      [after]: __after,
      [backdrop]: __backdrop,
      [before]: __before,
      [cue]: __cue,
      [firstLetter]: __firstLetter,
      [firstLine]: __firstLine,
      [placeholder]: __placeholder,
      [selection]: __selection,

      ...rest
    });
  },
);

Box.displayName = 'Box';

export default Box;
