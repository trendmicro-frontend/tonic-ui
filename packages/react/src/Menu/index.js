import Menu from './Menu';
import MenuList from './MenuList';
import MenuItem from './MenuItem';
import MenuGroup from './MenuGroup';
import MenuDivider from './MenuDivider';

Menu.List = MenuList;
Menu.Item = MenuItem;
Menu.Group = MenuGroup;
Menu.Divider = MenuDivider;

export {
  Menu,
  MenuList,
  MenuItem,
  MenuGroup,
  MenuDivider
};

export default Menu;
