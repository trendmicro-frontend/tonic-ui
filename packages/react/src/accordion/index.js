import Accordion from './Accordion';
import AccordionBody from './AccordionBody';
import AccordionCollapse from './AccordionCollapse';
import AccordionHeader from './AccordionHeader';
import AccordionItem from './AccordionItem';
import AccordionToggle from './AccordionToggle';
import AccordionToggleIcon from './AccordionToggleIcon';
import useAccordion from './useAccordion';
import useAccordionItem from './useAccordionItem';

Accordion.Body = AccordionBody;
Accordion.Collapse = AccordionCollapse;
Accordion.Header = AccordionHeader;
Accordion.Item = AccordionItem;
Accordion.Toggle = AccordionToggle;
Accordion.Toggle.Icon = AccordionToggleIcon;

export {
  Accordion,
  AccordionBody,
  AccordionCollapse,
  AccordionHeader,
  AccordionItem,
  AccordionToggle,
  AccordionToggleIcon,
  useAccordion,
  useAccordionItem,
};
