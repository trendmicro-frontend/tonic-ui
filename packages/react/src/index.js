// components
export * from './accordion';
export * from './alert';
export * from './badge';
export * from './box';
export * from './button';
export * from './checkbox';
export * from './color-mode';
export * from './color-style';
export * from './css-baseline';
export * from './divider';
export * from './drawer';
export * from './flat-button';
export * from './flex';
export * from './grid';
export * from './icon';
export * from './image';
export * from './input';
export * from './link';
export * from './menu';
export * from './modal';
export * from './pagination';
export * from './popover';
export * from './popper';
export * from './portal';
export * from './presence';
export * from './radio';
export * from './search-input';
export * from './scrollbar';
export * from './select';
export * from './skeleton';
export * from './space';
export * from './spinner';
export * from './stack';
export * from './switch';
export * from './table';
export * from './tabs';
export * from './tag';
export * from './text';
export * from './textarea';
export * from './theme';
export * from './toast';
export * from './tooltip';
export * from './transitions';
export * from './visually-hidden';

// hooks
export useDisclosure from './deprecated/useDisclosure'; // deprecated

// settings
export colorStyle from './shared/color-style';
export theme from './shared/theme';
