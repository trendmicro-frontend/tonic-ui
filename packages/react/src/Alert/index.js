import React, { forwardRef } from 'react';
import Box from '../Box';
import ButtonBase from '../ButtonBase';
import Flex from '../Flex';
import Icon from '../Icon';
import Space from '../Space';
import {
  useAlertStyle,
  useAlertIconStyle,
  useAlertMessageStyle,
  useAlertCloseButtonStyle,
} from './styles';

const defaultVariant = 'solid';
const defaultSeverity = 'success';

const getIconBySeverity = (severity) => {
  const iconName = {
    success: 'success',
    info: 'info',
    warning: 'warning-triangle',
    error: 'error',
  }[severity];

  if (!iconName) {
    return null;
  }

  return (
    <Icon icon={`${iconName}`} />
  );
};

const AlertIcon = (props) => (
  <Flex {...props} />
);

const AlertMessage = (props) => (
  <Box {...props} />
);

const AlertCloseButton = (props) => (
  <ButtonBase {...props} />
);

const Alert = forwardRef((
  {
    isClosable: _isClosable = false,
    isCloseButtonVisible: LEGACY_isCloseButtonVisible = false, // eslint-disable-line camelcase
    onClose,
    variant = defaultVariant,
    severity = defaultSeverity,
    icon,
    children,
    ...rest
  },
  ref,
) => {
  const isClosable = _isClosable || LEGACY_isCloseButtonVisible; // eslint-disable-line camelcase
  const styleProps = useAlertStyle({ variant, severity });
  const iconStyleProps = useAlertIconStyle({ variant, severity });
  const messageStyleProps = useAlertMessageStyle();
  const closeButtonStyleProps = useAlertCloseButtonStyle({ variant });

  if (typeof icon === 'string') {
    icon = (<Icon icon={icon} />);
  }
  if (typeof icon === 'undefined') {
    icon = getIconBySeverity(severity);
  }

  return (
    <Flex
      ref={ref}
      align="flex-start"
      justify="space-between"
      {...styleProps}
      {...rest}
    >
      {!!icon && (
        <>
          <AlertIcon {...iconStyleProps}>
            {icon}
          </AlertIcon>
          <Space minWidth="2x" />
        </>
      )}
      <AlertMessage {...messageStyleProps}>
        {children}
      </AlertMessage>
      {!!isClosable && (
        <>
          <Space minWidth="4x" />
          <AlertCloseButton {...closeButtonStyleProps} onClick={onClose}>
            <Icon icon="close-s" />
          </AlertCloseButton>
        </>
      )}
    </Flex>
  );
});

Alert.displayName = 'Alert';

export default Alert;
