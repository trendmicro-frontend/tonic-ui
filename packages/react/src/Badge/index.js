import React, { forwardRef } from 'react';
import useBadgeStyle from './styles';
import Box from '../Box';

const Badge = forwardRef(
  (
    {
      variant = 'badge',
      variantColor = 'red',
      badgeContent,
      isHidden = false,
      children,
      offset,
      dotSize = 8,
      borderColor,
      borderWidth = 1,
      ...restProps
    },
    ref,
  ) => {
    const badgeBorderColor = borderColor ?? variantColor;
    const badgeStyleProps = useBadgeStyle({
      color: variantColor,
      hasChildren: !!children,
      showAsDot: variant === 'dot',
      offset,
      dotSize,
      borderColor: badgeBorderColor,
      borderWidth
    });

    return (
      <Box
        as="span"
        position="relative"
        display="inline-block"
      >
        {children}
        {!isHidden && (
          <Box
            ref={ref}
            as="span"
            fontWeight="normal"
            borderRadius="sm"
            whiteSpace="nowrap"
            {...badgeStyleProps}
            {...restProps}
          >
            {badgeContent}
          </Box>
        )}
      </Box>
    );
  }
);

Badge.displayName = 'Badge';

export default Badge;
