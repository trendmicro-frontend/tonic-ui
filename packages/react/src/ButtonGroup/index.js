import memoize from 'micro-memoize';
import React, { forwardRef } from 'react';
import Box from '../Box';
import { ButtonGroupProvider } from './context';

const getMemoizedState = memoize(state => ({ ...state }));

const ButtonGroup = forwardRef((
  {
    children,
    size = 'md',
    variant = 'default',
    orientation = 'horizontal',
    ...rest
  },
  ref
) => {
  const buttonGroupState = getMemoizedState({ size, variant, orientation });
  const orientationProps = {
    vertical: {
      flexDirection: 'column',
    },
    horizontal: {
      flexDirection: 'row',
    },
  }[orientation];
  return (
    <ButtonGroupProvider value={buttonGroupState}>
      <Box
        ref={ref}
        display="inline-flex"
        {...orientationProps}
        {...rest}
      >
        {children}
      </Box>
    </ButtonGroupProvider>
  );
});

ButtonGroup.displayName = 'ButtonGroup';

export default ButtonGroup;
