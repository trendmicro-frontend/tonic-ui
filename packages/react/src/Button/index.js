import React, { forwardRef } from 'react';
import ButtonBase from '../ButtonBase';
import { useButtonGroup } from '../ButtonGroup/context';
import { getButtonGroupCSS, useButtonStyle } from './styles';

const defaultSize = 'md';
const defaultVariant = 'default';

const Button = forwardRef(
  (
    {
      as: Comp = 'button',
      type = 'button',
      borderRadius = 'sm',
      size,
      variant,
      children,
      css,
      ...rest
    },
    ref,
  ) => {
    const buttonGroupContext = useButtonGroup();
    let isInGroup = false;
    let useVertical = false;
    if (buttonGroupContext) {
      isInGroup = true;
      const {
        size: buttonGroupSize,
        variant: buttonGroupVariant,
        orientation,
      } = { ...buttonGroupContext };
      useVertical = (orientation === 'vertical');
      // - Use the inherited value from the button group
      // - Fallback to the default value if the value is null or undefined
      size = (buttonGroupSize ?? size) ?? defaultSize;
      variant = (buttonGroupVariant ?? variant) ?? defaultVariant;
    } else {
      // Use the default value if the value is null or undefined
      size = size ?? defaultSize;
      variant = variant ?? defaultVariant;
    }
    const buttonStyleProps = useButtonStyle({
      size,
      variant,
      borderRadius,
    });
    css = [
      isInGroup && getButtonGroupCSS({ useVertical }),
      { ...css }
    ];

    return (
      <ButtonBase
        ref={ref}
        as={Comp}
        type={type}
        borderRadius={borderRadius}
        css={css}
        {...buttonStyleProps}
        {...rest}
      >
        { children }
      </ButtonBase>
    );
  },
);

Button.displayName = 'Button';

export default Button;
