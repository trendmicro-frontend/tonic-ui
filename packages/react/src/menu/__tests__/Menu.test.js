import { act, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { render } from '@tonic-ui/react/test-utils/render';
import { testA11y } from '@tonic-ui/react/test-utils/accessibility';
import { Menu, MenuButton, MenuList, MenuItem } from '@tonic-ui/react/src';
import React from 'react';

describe('Menu', () => {
  it('should render correctly', async () => {
    const user = userEvent.setup();
    const items = [
      'Menu item 1',
      'Menu item 2',
      'Menu item 3',
    ];
    const { baseElement, container } = render(
      <Menu>
        <MenuButton variant="secondary" data-testid="button">
          Open
        </MenuButton>
        <MenuList>
          {items.map((item) => (
            <MenuItem key={item}>
              {item}
            </MenuItem>
          ))}
        </MenuList>
      </Menu>
    );

    const button = screen.getByTestId('button');
    expect(button).toBeInTheDocument();

    await act(() => user.click(button));

    expect(await screen.findByRole('menu')).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();

    await testA11y(container);
  });
});
