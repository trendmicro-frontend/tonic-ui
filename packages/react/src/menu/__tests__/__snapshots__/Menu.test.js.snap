// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: inherit;
  border: var(--tonic-borders-1);
  color: var(--tonic-colors-white-primary);
  cursor: pointer;
  outline: 0;
  padding: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  border-radius: var(--tonic-radii-sm);
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  -webkit-transition: background-color 200ms cubic-bezier(0.4, 0, 0.2, 1),border-color 200ms cubic-bezier(0.4, 0, 0.2, 1),box-shadow 200ms cubic-bezier(0.4, 0, 0.2, 1),color 200ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: background-color 200ms cubic-bezier(0.4, 0, 0.2, 1),border-color 200ms cubic-bezier(0.4, 0, 0.2, 1),box-shadow 200ms cubic-bezier(0.4, 0, 0.2, 1),color 200ms cubic-bezier(0.4, 0, 0.2, 1);
  min-height: var(--tonic-sizes-8x);
  font-size: var(--tonic-fontSizes-sm);
  line-height: var(--tonic-lineHeights-sm);
  border-color: var(--tonic-colors-gray-60);
  -webkit-column-gap: var(--tonic-sizes-1x);
  column-gap: var(--tonic-sizes-1x);
}

.emotion-2[aria-selected=true],
.emotion-2[data-selected] {
  background-color: var(--tonic-colors-gray-70);
  color: var(--tonic-colors-white-emphasis);
  pointer-events: none;
}

.emotion-2:focus,
.emotion-2[data-focus] {
  border-color: var(--tonic-colors-blue-60);
  box-shadow: inset 0 0 0 .0625rem #1e5ede;
  color: var(--tonic-colors-white-primary);
  z-index: 1;
}

.emotion-2:hover,
.emotion-2[data-hover] {
  color: var(--tonic-colors-blue-40);
  z-index: 2;
}

.emotion-2:hover:not(:focus),
.emotion-2[data-hover]:not(:focus) {
  border-color: var(--tonic-colors-blue-50);
}

.emotion-2:active,
.emotion-2[data-active] {
  border-color: var(--tonic-colors-blue-50);
  background-color: rgba(0, 0, 0, 0.12);
  color: var(--tonic-colors-blue-40);
}

.emotion-2:disabled,
.emotion-2:disabled:focus,
.emotion-2:disabled:hover,
.emotion-2[aria-disabled=true],
.emotion-2[aria-disabled=true]:focus,
.emotion-2[aria-disabled=true]:hover,
.emotion-2[data-disabled] {
  border-color: var(--tonic-colors-gray-60);
  color: var(--tonic-colors-white-emphasis);
  cursor: not-allowed;
  opacity: 0.28;
}

.emotion-2:hover {
  color: var(--tonic-colors-white-primary);
}

.emotion-2:active {
  color: var(--tonic-colors-white-primary);
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transform: rotate(-180deg);
  -moz-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
  -webkit-transition: -webkit-transform 133ms cubic-bezier(0.0, 0, 0.2, 1);
  transition: transform 133ms cubic-bezier(0.0, 0, 0.2, 1);
}

.emotion-6:disabled,
.emotion-6:disabled:focus,
.emotion-6:disabled:hover,
.emotion-6[aria-disabled=true],
.emotion-6[aria-disabled=true]:focus,
.emotion-6[aria-disabled=true]:hover,
.emotion-6[data-disabled] {
  opacity: .28;
}

.emotion-8 {
  width: var(--tonic-sizes-4x);
  height: var(--tonic-sizes-4x);
  fill: currentColor;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-8:not(:root) {
  overflow: hidden;
}

.emotion-10 {
  position: absolute;
  z-index: var(--tonic-zIndices-dropdown);
  outline: 0;
  background-color: var(--tonic-colors-gray-80);
  box-shadow: 0 4px 16px 0 rgba(0, 0, 0, 0.48),0 2px 4px 0 rgba(0, 0, 0, 0.16);
  color: inherit;
  margin: 0;
  padding: 0;
  padding-top: var(--tonic-space-2x);
  padding-bottom: var(--tonic-space-2x);
}

.emotion-12 {
  height: 0;
  opacity: 1;
  overflow: hidden;
  -webkit-transition: height 133ms linear,opacity 133ms linear;
  transition: height 133ms linear,opacity 133ms linear;
}

.emotion-16 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: inherit;
  border: none;
  color: var(--tonic-colors-white-primary);
  cursor: pointer;
  outline: 0;
  padding: 0;
  font-size: var(--tonic-fontSizes-sm);
  line-height: var(--tonic-lineHeights-sm);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  padding-left: var(--tonic-space-3x);
  padding-right: var(--tonic-space-3x);
  padding-top: var(--tonic-space-2x);
  padding-bottom: var(--tonic-space-2x);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 100%;
}

.emotion-16[aria-selected=true],
.emotion-16[data-selected] {
  background-color: rgba(255, 255, 255, 0.08);
}

.emotion-16:focus,
.emotion-16[data-focus] {
  border-color: var(--tonic-colors-blue-60);
  border-style: solid;
  border-width: var(--tonic-sizes-1h);
  padding-left: calc(.75rem - .125rem);
  padding-right: calc(.75rem - .125rem);
  padding-top: calc(.5rem - .125rem);
  padding-bottom: calc(.5rem - .125rem);
}

.emotion-16:hover,
.emotion-16[data-hover] {
  background-color: rgba(255, 255, 255, 0.12);
}

.emotion-16:disabled,
.emotion-16:disabled:focus,
.emotion-16:disabled:hover,
.emotion-16[aria-disabled=true],
.emotion-16[aria-disabled=true]:focus,
.emotion-16[aria-disabled=true]:hover,
.emotion-16[data-disabled] {
  background-color: inherit;
  color: var(--tonic-colors-white-disabled);
  cursor: not-allowed;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <button
      aria-controls="@tonic-ui/react:Menu-1"
      aria-expanded="true"
      aria-haspopup="menu"
      class="emotion-2 emotion-1"
      data-testid="button"
      id="@tonic-ui/react:MenuToggle-1"
      role="button"
      tabindex="0"
      type="button"
    >
      <div
        class="emotion-4 emotion-1"
      >
        Open
      </div>
      <div
        class="emotion-6 emotion-1"
      >
        <svg
          aria-hidden="true"
          class="emotion-8 emotion-9"
          focusable="false"
          role="presentation"
          viewBox="0 0 16 16"
        >
          <g>
            <path
              d="M12.5 6.5l-4.5 4.5-4.5-4.5 1-1 3.5 3.5 3.5-3.5 1 1z"
            />
          </g>
        </svg>
      </div>
    </button>
    <div
      aria-labelledby="@tonic-ui/react:MenuToggle-1"
      class="emotion-10 emotion-1"
      data-menu-id="@tonic-ui/react:Menu-1"
      data-popper-escaped=""
      data-popper-placement="bottom-start"
      data-popper-reference-hidden=""
      data-testid="menu-list"
      id="@tonic-ui/react:Menu-1"
      role="menu"
      style="position: absolute; left: 0px; top: 0px; margin: 0px; transform: translate(0px, 0px);"
      tabindex="-1"
    >
      <div
        class="emotion-12 emotion-1"
      >
        <div
          class="emotion-4 emotion-1"
        >
          <button
            class="emotion-16 emotion-1"
            data-id="1"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Menu item 1
          </button>
          <button
            class="emotion-16 emotion-1"
            data-id="2"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Menu item 2
          </button>
          <button
            aria-disabled="true"
            class="emotion-16 emotion-1"
            data-id="3"
            disabled=""
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            Menu item 3
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
